src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Registry/PluginRegistry.cs

```diff
@@ -198,11 +347,26 @@ private async Task<List<InstalledPluginInfo>> ReadInstalledPlugins()
             {
                 return new List<InstalledPluginInfo>();
             }
-            
-            using (var stream = new FileStream(this.registryFilePath, FileMode.Open, FileAccess.Read))
+
+            try
+            {
+                using (var fileStream = new FileStream(this.registryFilePath, FileMode.Open, FileAccess.Read, FileShare.Read))
+                {
+                    return await JsonSerializer.DeserializeAsync<List<InstalledPluginInfo>>(
+                        fileStream,
+                        SerializationConfig.PluginsManagerSerializerDefaultOptions);
+                }
+            }
+            catch (Exception e)
```

Comment:
I don't think this should be a catch-all. There's probably only a subset of exceptions we expect to happen here

Reply:
The catch is basically to wrap the exception happens during the read and write of plugin registry file in a custom type and re-throw.  Would you prefer system exceptions like `UnauthorizedAccessException` to not be wrapped? Maybe only expect JsonException and IOException?